actors/miner/src/expiration_queue.rs

```diff
@@ -370,49 +370,45 @@ impl<'db, BS: Blockstore> ExpirationQueue<'db, BS> {
         let mut sectors_rescheduled = Vec::<&SectorOnChainInfo>::new();
         let mut recovered_power = PowerPair::zero();
 
-        self.iter_while_mut(|_epoch, expiration_set| {
-            let on_time_sectors: BTreeSet<SectorNumber> = expiration_set
-                .on_time_sectors
-                .bounded_iter(ENTRY_SECTORS_MAX)
-                .context("too many sectors to reschedule")?
-                .map(|i| i as SectorNumber)
-                .collect();
+        let mut sectors_field = BitField::new();
+        let mut sectors_by_number = BTreeMap::<u64, &SectorOnChainInfo>::new();
 
-            let early_sectors: BTreeSet<SectorNumber> = expiration_set
-                .early_sectors
-                .bounded_iter(ENTRY_SECTORS_MAX)
-                .context("too many sectors to reschedule")?
-                .map(|i| i as SectorNumber)
-                .collect();
+        for sector in &sectors {
+            sectors_field.set(sector.sector_number);
+            sectors_by_number.insert(sector.sector_number, sector);
+        }
+
+        self.iter_while_mut(|_epoch, expiration_set| {
+            let on_time_sectors = &sectors_field & &expiration_set.on_time_sectors;
```

Comment:
pre-grouping by expiration time will likely be _much_ faster.

Reply:
> pre-grouping by expiration time will likely be _much_ faster.

Yes, we discussed it yesterday, and will refine the patch and test on mainnet. Thanks for figuring out.